OpenRA.Mods.Common/UtilityCommands/CheckExplicitInterfacesCommand.cs

```diff
@@ -0,0 +1,120 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Linq;
+using System.Reflection;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.UtilityCommands
+{
+	public class CheckExplicitInterfacesCommand : IUtilityCommand
+	{
+		public string Name { get { return "--check-explicit-interfaces"; } }
+
+		bool IUtilityCommand.ValidateArguments(string[] args) { return true; }
+
+		int violationCount;
+
+		[Desc("Check for explicit interface implementation violations in all OpenRA assemblies referenced by the specified mod.")]
+		void IUtilityCommand.Run(ModData modData, string[] args)
+		{
+			var types = AppDomain.CurrentDomain.GetAssemblies()
+				.Where(asm => asm.FullName.Contains("OpenRA"))
+				.SelectMany(asm => asm.GetTypes());
+
+			foreach (var implementingType in types.Where(t => !t.IsInterface))
+			{
+				if (implementingType.IsEnum)
+					continue;
+
+				var interfaces = implementingType.GetInterfaces();
+				foreach (var interfaceType in interfaces)
+				{
+					if (!interfaceType.HasAttribute<RequireExplicitImplementationAttribute>())
+						continue;
+
+					if (!interfaceType.Namespace.Contains("OpenRA"))
```

Comment:
Good point.
This existed because before the attrib was implemented Eluant wasn't playing nice.